pkg/scheduler/core/equivalence_cache.go

```diff
@@ -86,6 +87,12 @@ func (ec *EquivalenceCache) RunPredicate(
 ) (bool, []algorithm.PredicateFailureReason, error) {
 	ec.mu.Lock()
 	defer ec.mu.Unlock()
+
+	if nodeInfo == nil || nodeInfo.Node() == nil {
```

Comment:
@resouer - do we know why this is nil only sometimes ?

Reply:
I have some theory, but would like to verify it somehow tomorrow :/

Reply:
In some of our tests, nodeInfo.Node() is nil. It is expected.